stl/inc/xstring

```diff
@@ -3167,17 +3169,17 @@ public:
 
                 if (_Right._Mypair._Myval2._Large_string_engaged()) {
                     const auto _New_size     = _Right._Mypair._Myval2._Mysize;
-                    const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());
+                    auto _New_capacity       = _Calculate_growth(_New_size, 0, _Right.max_size()) + 1;
                     auto _Right_al_non_const = _Right_al;
-                    const auto _New_ptr      = _Right_al_non_const.allocate(_New_capacity + 1); // throws
+                    const auto _New_ptr      = _Allocate_at_least_helper(_Right_al_non_const, _New_capacity); // throws
 
-                    _Start_element_lifetimes(_Unfancy(_New_ptr), _New_size + 1);
+                    _Start_element_lifetimes(_Unfancy(_New_ptr), _New_capacity);

```

Comment:
I'm fairly certain using `_New_size + 1` here previously was a bug, can anyone confirm?

Reply:
`basic_string` always has a null terminator at the end of the sequence which isn't included in the size.

Reply:
But even before my change it was allocating `_New_capacity + 1` elements while only initializing `_New_size + 1` elements. At all other source locations where `_New_capacity + 1` elements are allocated the same amount is also initialized.

Reply:
Ah! I was reading this as "I think '`+ 1`' is a bug" and not your intended "I think 'size' instead of 'capacity' is a bug", which I agree with. This should be `_New_capacity + 1` as it is everywhere else.